lib/generators/shopify_app/add_marketing_activity_extension/add_marketing_activity_extension_generator.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails/generators/base'
+
+module ShopifyApp
+  module Generators
+    class AddMarketingActivityExtensionGenerator < Rails::Generators::Base
+      source_root File.expand_path('../templates', __FILE__)
+
+      def generate_app_extension
+        generate "shopify_app:hmac_controller"
+
+        template "marketing_activities_controller.rb", "app/controllers/marketing_activities_controller.rb"
+        generate_routes
+      end
+
+      private
+
+      def generate_routes
+        inject_into_file(
+          'config/routes.rb',
+          optimize_indentation(routes, 2),
+          after: "root :to => 'home#index'\n"
+        )
+      end
+
+      def routes
+        <<~EOS
+          \nresource :marketing_activities, only: [:create, :update] do
```

Comment:
why do you have a newline at the beginning of this rather than just putting a blank line in  your template?